### DISCUSSION TITLE
Wildcard validation does not work ?

### DISCUSSION THREAD

**Discussion by jeanali** [livewire/livewire#3025] [7. Bugs (archived)]
I'm facing this error with "Wildcard Validation", I have an array like this:

```langs => [0=>'en',1=>'le'];```

I have Foreach of langs with input like this:

```
<form wire:submit.prevent="store">
  @foreach ($langs as $index => $lang)
      <div wire:key="languages-{{ $index }}">
          <label>{{ $lang }}</label>
          <input type="text" wire:model="name.{{ $lang }}" />
          @error('name.'.$lang) <span class="error">{{ $message }}</span> @enderror
      </div>
  @endforeach
  <button type="submit" class="btn py-2 px-4 mt-7">
      @lang('site.save')
  </button>
</form>
```
Livewire component:

```
public $name = [];

protected $rules = [
    'name.*' => 'required|string|min:5|max:100',
];

public function store()
{
    $this->validate();

    dd('run');
} 
```

When I just submit without putting any values in input the validation does not work! but when I fill the first input then submit the form the second input validation worked!

Why when no values validation does not work!

**Comment by joshhanley** [comment]
@jeanali you need to use the `$index` as part of your property name, not the value.

Try this
```blade
<form wire:submit.prevent="store">
  @foreach ($langs as $index => $lang)
      <div wire:key="languages-{{ $index }}">
          <label>{{ $lang }}</label>
          <input type="text" wire:model="name.{{ $index }}" />
          @error('name.'.$index) <span class="error">{{ $message }}</span> @enderror
      </div>
  @endforeach
  <button type="submit" class="btn py-2 px-4 mt-7">
      @lang('site.save')
  </button>
</form>
```

Hope this helps!

**Comment by jeanali** [comment]
@joshhanley  Now I got this error: 

```
Column not found: 1054 Unknown column 'name.0' in 'where clause' 
```

What should I change in validation?!

**Comment by joshhanley** [comment]
@jeanali well that's good in a way, as that's a different problem I think! 

The validation you have shared doesn't have any code that would touch the database, so there is something else going on.

Can you share you full component class, as I suspect there is a database query not setup right or something?

Thanks
